diff --git a/arch/arm/boot/dts/aspeed/aspeed-g6.dtsi b/arch/arm/boot/dts/aspeed/aspeed-g6.dtsi
index 7fb421153596..09fcef6d999c 100644
--- a/arch/arm/boot/dts/aspeed/aspeed-g6.dtsi
+++ b/arch/arm/boot/dts/aspeed/aspeed-g6.dtsi
@@ -332,6 +332,25 @@ apb {
 			#size-cells = <1>;
 			ranges;
 
+			pwm_tach: pwm-tacho-controller@1e610000 {
+				compatible = "aspeed,ast2600-pwm-tach";
+				reg = <0x1e610000 0x100>;
+				clocks = <&syscon ASPEED_CLK_AHB>;
+				resets = <&syscon ASPEED_RESET_PWM>;
+				#pwm-cells = <3>;
+				status = "disabled";
+
+				fan-0 {
+					tach-ch = <0x0>;
+					pwms = <&pwm_tach 0 40000 0>;
+				};
+
+				fan-1 {
+					tach-ch = <0x1 0x2>;
+					pwms = <&pwm_tach 1 40000 0>;
+				};
+			};
+
 			hace: crypto@1e6d0000 {
 				compatible = "aspeed,ast2600-hace";
 				reg = <0x1e6d0000 0x200>;
